tests/test_series.py

```diff
@@ -3543,6 +3543,14 @@ def test_series_add_complex() -> None:
     s = pd.Series([1.0, 2.0, 3.0])
     check(assert_type(s + c, "pd.Series[complex]"), pd.Series)
 
+    c1 = 1
+    s1 = pd.Series([1.0, 2.0, 3.0])
+    check(assert_type(s1 + c1, "pd.Series[float]"), pd.Series)
+
+    c2 = 1
+    s2 = pd.Series([1, 2, 3])
+    check(assert_type(s2 + c2, "pd.Series[int]"), pd.Series)
```

Comment:
You also need to add a test for when the `Series` is `Series[Any]`.  This is where it gets difficult to get this right:
```python
s3 = pd.DataFrame({"a": [1,2,3]})["s3"]  # This will resolve to `Series[Unknown]` in `pyright` and `Series[Any]` in `mypy`
check(assert_type(s3 + 2, "pd.Series"), pd.Series)
check(assert_type(s3 + 2.2, "pd.Series"), pd.Series)
check(assert_type(s3 + (1 + 2j), "pd.Series"), pd.Series)
```